src/interface/hooks/use-post-present-proof-send-proposal.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * This function is used to send with POST a proposal for a present proof
+ */
+import { useCallback, useState } from 'react'
+import post from '../api/helpers/post'
+
+export default function usePostPresentProofSendProposal() {
+  const path = '/present-proof-2.0/send-proposal'
+  const transformData = (retrievedData) => {
+    return retrievedData.pres_ex_id
+  }
+  const [error, setError] = useState(null)
+
+  const onStartFetch = useCallback(
+    (origin, connectionId, id, type, credDefId, setStatus, setStoreData) => {
+      const createBody = (connectionId, id, type, credDefId) => {
```

Comment:
would it be possible to move this function outside of hook `onStartFetch` method so it's easier to read hook? is because that this look like an validation helper/util

Reply:
done, will make a similar change in some other hooks that have the same problem